test/fail_compilation/bug9631.d

```diff
@@ -1,21 +1,32 @@
 /*
 TEST_OUTPUT:
 ---
-fail_compilation/bug9631.d(20): Error: cannot implicitly convert expression `F()` of type `bug9631.T1!().F` to `bug9631.T2!().F`
+fail_compilation/bug9631.d(17): Error: cannot implicitly convert expression `F()` of type `bug9631.tem!().F` to `bug9631.F`
 ---
 */
 
-template T1()
+struct F { }
+
+template tem()
 {
     struct F { }
 }
 
-template T2()
+void convert()
 {
-    struct F { }
+    F x = tem!().F();
 }
 
-void main()
```

Comment:
Why the refactoring of this test case.  There should only be changes to the error messages.

Reply:
I thought it was a bit simpler whilst still essentially testing the same thing. But I suppose it may be worth having both the two template version and the new changes.

Reply:
Restored the two template version.